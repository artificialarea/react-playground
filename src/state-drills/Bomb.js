import React from 'react';

// src: https://courses.thinkful.com/react-v1/checkpoint/9#assignment
// alt solution: https://github.com/tomatau/state-drills/blob/master/src/state-drills/Bomb.js

class Bomb extends React.Component {
  
  constructor(props) {
    super(props)
    this.state = {
      boom: '"BOOM!!!!"',
      count: 1,
      tick: '"tick"',
      tock: '"tock"',
    }
  }

  componentDidMount() {
    this.interval = setInterval(() => {
      const newCount = this.state.count + 1
      this.setState(
        { count: newCount }
      )
    }, 1000)
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  render() {
    const counter = this.state.count
    let sound = counter % 2 === 0 ? this.state.tock : this.state.tick

    if (counter >= 10 ) {
      sound = this.state.boom
      this.componentWillUnmount()
    }

    return (
      <div>
        <p className="sound">{sound}</p>
        <p className="counter">{counter}</p>
        
      </div>
    )
  }
}

export default Bomb;
