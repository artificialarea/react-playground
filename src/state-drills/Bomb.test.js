import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Bomb from './Bomb';

describe('<Bomb />', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Bomb />, div)
    ReactDOM.unmountComponentAtNode(div)
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(<Bomb />)
      .toJSON()
    expect(tree).toMatchSnapshot();
  });

});