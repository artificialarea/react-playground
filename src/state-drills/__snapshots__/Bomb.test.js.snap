// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bomb /> renders UI as expected 1`] = `
<div>
  <p
    className="sound"
  >
    "tick"
  </p>
  <p
    className="counter"
  >
    1
  </p>
</div>
`;
